controllers/spacebindingrequest/spacebindingrequest_controller.go

```diff
@@ -209,19 +235,20 @@ func (r *Reconciler) addFinalizer(ctx context.Context, memberCluster cluster.Clu
 	return nil
 }
 
-func (r *Reconciler) ensureSpaceBinding(ctx context.Context, memberCluster cluster.Cluster, spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest, spaceBinding *toolchainv1alpha1.SpaceBinding) error {
+func (r *Reconciler) ensureSpaceBinding(ctx context.Context, memberCluster cluster.Cluster, spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) error {
 	logger := log.FromContext(ctx)
 	logger.Info("ensuring spacebinding")
+	// create spacebinding if not found for given spaceBindingRequest
+	spaceBinding, err := r.getSpaceBinding(ctx, spaceBindingRequest, memberCluster)
+	if err != nil {
+		return err
+	}
 
 	// find space from namespace labels
 	space, err := r.getSpace(ctx, memberCluster, spaceBindingRequest)
 	if err != nil {
 		return err
 	}
```

Comment:
the `getSpace` is called also at the very beginning of the `getSpaceBinding` method. so you could swap it and drop the extra call from `getSpaceBinding`.
```suggestion
	// find space from namespace labels
	space, err := r.getSpace(ctx, memberCluster, spaceBindingRequest)
	if err != nil {
		return err
	}

	// create spacebinding if not found for given spaceBindingRequest
	spaceBinding, err := r.getSpaceBinding(ctx, spaceBindingRequest, space)
	if err != nil {
		return err
	}
```

Reply:
Good point!  Thanks!

I've opted for returning also the space object from `getSpaceBinding` in https://github.com/codeready-toolchain/host-operator/pull/898/commits/c848d1e49b3fc3972bface45c27b2ad335ff894b

Mainly because I'm calling `getSpaceBinding` also from `ensureSpaceBindingDeletion` as one of the first things at the beginning of the `reconcile` function , and I don't have the space there yet. I think it's handy to retrieve the space from within  `getSpaceBinding`. The other option would be to call both `getSpace` and `getSpaceBinding` from the `reconcile` and pass the `spacebinding` and `space` objects to `ensureSpaceBindingDeletion` or `ensureSpaceBinding`. But this might require some reshuffle of the tests. 

Please let me know WDYT.